common/signal_processing/src/butterworth.cpp

```diff
@@ -0,0 +1,295 @@
+   * */
+  if (fc >= fs / 2) {
+    print("Invalid argument : Cut-off frequency  fc must be less than fs/2 \n");
+    return;
+  }
+
+  cutoff_frequency_rad_sec = fc * 2.0 * M_PI;
+  sampling_frequency_hz = fs;
+}
+
+sOrderCutoff ButterworthFilter::getOrderCutOff() const
+{
+  return sOrderCutoff{order_, cutoff_frequency_rad_sec};
+}
+
+/**
+ * @brief Matlab equivalent : [b, a]  = butter(n, Wn, 's')
+ * */
+void ButterworthFilter::computeContinuousTimeTF(const bool & use_sampling_frequency)
+{
+  // First compute  the phase angles of the roots
+  computePhaseAngles();
+  computeContinuousTimeRoots(use_sampling_frequency);
+
+  continuous_time_denominator_ = poly(continuous_time_roots_);
+  continuous_time_numerator_ = std::pow(cutoff_frequency_rad_sec, order_);
+}
+
+void ButterworthFilter::computePhaseAngles()
+{
+  phase_angles_.resize(order_, 0.);
+  int k{1};
+
+  for (auto & x : phase_angles_) {
+    x = M_PI_2 + (M_PI * (2.0 * k - 1.0) / (2.0 * order_));
+    k++;
+  }
+}
+
+void ButterworthFilter::computeContinuousTimeRoots(const bool & use_sampling_frequency)
+{
+  continuous_time_roots_.resize(order_, {0.0, 0.0});
+  int k{};
+
+  if (use_sampling_frequency) {
+    double const & Fc = (sampling_frequency_hz / M_PI) *
+                        tan(cutoff_frequency_rad_sec / (sampling_frequency_hz * 2.0));
+
+    for (auto & x : continuous_time_roots_) {
+      x = {
+        std::cos(phase_angles_[k]) * Fc * 2.0 * M_PI, std::sin(phase_angles_[k]) * Fc * 2.0 * M_PI};
+      k++;
+    }
+
+  } else {
+    for (auto & x : continuous_time_roots_) {
+      x = {
+        cutoff_frequency_rad_sec * cos(phase_angles_[k]),
+        cutoff_frequency_rad_sec * sin(phase_angles_[k])};
+      k++;
+    }
+  }
+}
+std::vector<std::complex<double>> ButterworthFilter::poly(
+  std::vector<std::complex<double>> const & roots)
+{
+  std::vector<std::complex<double>> coefficients(roots.size() + 1, {0, 0});
+
+  int const n{static_cast<int>(roots.size())};
+
+  coefficients[0] = {1.0, 0.0};
+
+  for (int i = 0; i < n; i++) {
+    for (int j = i; j != -1; j--) {
+      coefficients[j + 1] = coefficients[j + 1] - (roots[i] * coefficients[j]);
+    }
+  }
+
+  return coefficients;
+}
+
+/**
+ *  @brief Prints the order and cut-off angular frequency (rad/sec) of the filter
+ * */
+
+void ButterworthFilter::printFilterContinuousTimeRoots() const
+{
+  print("\n The roots of the continuous-time filter Transfer Function's Denominator are : \n");
+
+  for (auto const & x : continuous_time_roots_) {
+    print(std::real(x), std::imag(x) < 0 ? " - " : " + ", " j", std::abs(std::imag(x)));
+  }
+}
+void ButterworthFilter::printContinuousTimeTF() const
+{
+  auto const & n = order_;
+
+  std::cout << "\nThe Continuous Time Transfer Function of the Filter is ;\n" << std::endl;
+
+  print(continuous_time_numerator_, " / ");
+
+  for (int i = n; i > 0; i--) {
+    printf("%4.3f *", continuous_time_denominator_[n - i].real());
+    printf("s[%d] + ", i);
```

Comment:
You are using std::cout and printf. This should be organized.
Better to use one of them or use RCLCPP_INFO